src/test/java/org/jenkinsci/plugins/ewm/steps/ExwsStepTest.java

```diff
@@ -0,0 +1,294 @@
+        j.assertLogContains(format("The Node 'node-one' config does not have defined any Disk Ref ID '%s'", selectedDisk.getDiskId()), run);
+    }
+
+    @Test
+    public void missingLocalRootPathFromNodeProperty() throws Exception {
+        setExternalWorkspaceNodeProperty(node1, DISK_POOL_ID, new DiskNode(DISK_ID_ONE, ""));
+
+        runWorkflowJob(job);
+
+        Disk selectedDisk = findAllocatedDisk(disk1, disk2);
+        j.assertBuildStatus(FAILURE, run);
+        j.assertLogContains(format("The Node 'node-one' config does not have defined any local root path for Disk Ref ID '%s'", selectedDisk.getDiskId()), run);
+    }
+
+    @Test
+    public void sharedWorkspaceBetweenTwoDifferentNodes() throws Exception {
+        setExternalWorkspaceNodeProperty(node1, DISK_POOL_ID, diskNode1, diskNode2);
+        setExternalWorkspaceNodeProperty(node2, DISK_POOL_ID, diskNode1, diskNode2);
+
+        WorkflowJob jobWithTwoNodes = createWorkflowJobWithTwoNodes();
+        runWorkflowJob(jobWithTwoNodes);
+        Disk allocatedDisk = findAllocatedDisk(disk1, disk2);
+
+        j.assertBuildStatusSuccess(run);
+        j.assertLogContains("Searching for disk definitions in the External Workspace Templates from Jenkins global config", run);
+        j.assertLogContains("Searching for disk definitions in the Node config", run);
+        j.assertLogContains(format("Running in %s/%s/%s/%d",
+                allocatedDisk.getMasterMountPoint(), allocatedDisk.getPhysicalPathOnDisk(), run.getParent().getName(), run.getNumber()), run);
+        // The text written to file should be printed twice on the console output (when writing and when reading the file)
+        assertThat(countMatches(JenkinsRule.getLog(run), TEXT), is(2));
+    }
+
+    @Test
+    public void sharedWorkspaceBetweenTwoDifferentNodesWithTemplate() throws Exception {
+        Template linuxTemplate = new Template(DISK_POOL_ID, "linux", Arrays.asList(diskNode1, diskNode2));
+        Template testTemplate = new Template(DISK_POOL_ID, "test", Arrays.asList(diskNode1, diskNode2));
+        setUpTemplates(linuxTemplate, testTemplate);
+
+        WorkflowJob jobWithTwoNodes = createWorkflowJobWithTwoNodes();
+        runWorkflowJob(jobWithTwoNodes);
+        Disk allocatedDisk = findAllocatedDisk(disk1, disk2);
+
+        j.assertBuildStatusSuccess(run);
+        j.assertLogContains("Searching for disk definitions in the External Workspace Templates from Jenkins global config", run);
+        j.assertLogNotContains("Searching for disk definitions in the Node config", run);
+        j.assertLogContains(format("Running in %s/%s/%s/%d",
+                allocatedDisk.getMasterMountPoint(), allocatedDisk.getPhysicalPathOnDisk(), run.getParent().getName(), run.getNumber()), run);
+        // The text written to file should be printed twice on the console output (when writing and when reading the file)
+        assertThat(countMatches(JenkinsRule.getLog(run), TEXT), is(2));
+    }
+
```

Comment:
I can think of one more test: a test with two different workflow jobs using the same workspace. Not so much to prove that it works, but to prevent future modifications to the plugin from breaking it without noticing.